main.go

```diff
@@ -128,6 +163,42 @@ func main() {
 	}
 }
 
+func (app *App) sessionMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		// set the cookies on the ory client
+		var cookies string
+
+		// this example passes all request.Cookies
+		// to `ToSession` function
+		//
+		// However, you can pass only the value of
+		// ory_session_projectid cookie to the endpoint
+		cookies = request.Header.Get("Cookie")
+		// check if we have a session
+		session, _, err := app.ory.FrontendApi.ToSession(request.Context()).Cookie(cookies).Execute()
+		if err != nil && session == nil {
+			// Render a separate page with a button to redirect the user to the login page
+			writer.WriteHeader(http.StatusOK)
+			writer.Header().Set("Content-Type", "text/html; charset=utf-8")
+			fmt.Fprint(writer, "<html><body>")
+			fmt.Fprint(writer, "<h1>No Ory Network Session found</h1>")
+			fmt.Fprint(writer, "<p>To access the Ory Community Slack, please sign in with your Ory account to receive your invite. You will be redirected back to https://slack.ory.sh/ after logging in.</p>")
+			fmt.Fprint(writer, "<form action=\"https://auth.slackinviter.vinckr.com/ui/login?return_to=https://slackinviter.vinckr.com/\" method=\"GET\">")
+			fmt.Fprint(writer, "<input type=\"submit\" value=\"To Ory Network Login\">")
+			fmt.Fprint(writer, "</form>")
+			fmt.Fprint(writer, "</body></html>")
+			return
```

Comment:
That's ok but you could also copy&paste this file https://github.com/ory/slackinviter/blob/master/templates/index.tmpl and modify it to show what you want to show!

Reply:
done! using redirect.tmpl now and copied the styles from index more or less.